devito/finite_differences/coefficients.py

```diff
@@ -224,6 +224,33 @@ def generate_subs(i):
         rules = {}
         for i in self.coefficients:
             rules.update(generate_subs(i))
+        return rules
+
+    def update_rules(self, obj):
```

Comment:
I don't like the word "update" because it makes one think you're changing the internal state of the object but (thanks God) you aren't

Reply:
Any suggestions for a replacement? 'bend_rules', 'modify_rules', 'adapt_rules'? I personally prefer the latter